components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/TypeConversionHandler.scala

```diff
@@ -86,6 +96,35 @@ object TypeConversionHandler {
       handleLooseNumberConversion(boxedGivenClass, boxedSuperclassCandidate)
   }
 
+  private def handleIndexedRecordToMapConversion(
+      from: SingleTypingResult,
+      to: TypedClass,
+      strict: Boolean
+  ): Boolean = {
+    if (!AssignabilityUtil.isAssignableToLoadableClass(
+        from.runtimeObjType.klass,
+        "org.apache.avro.generic.IndexedRecord"
+      ) ||
+      !ClassUtils.isAssignable(to.runtimeObjType.klass, classOf[java.util.Map[_, _]])) {
+      false
+    } else {
+      val indexedRecordKeyParam = Typed.genericTypeClass(classOf[String], List())
+      val indexedRecordValueParam = from match {
+        case fromCasted: TypedObjectTypingResult =>
+          superTypeOfTypes(fromCasted.fields.values)
+        case _ => Unknown
+      }
+      val isAssignable =
```

Comment:
Please rebase to my change: #7984 . In this change, conversion checking strategy is passed implicitly, so there is no need to recover it from boolean, which was computed based on strategy which was computed based on boolean ;)